pkg/kubelet/kuberuntime/logs/logs.go

```diff
@@ -354,10 +355,15 @@ func ReadLogs(ctx context.Context, path, containerID string, opts *LogOptions, r
 				if err != nil {
 					return err
 				}
+				if tryrecreated {
```

Comment:
```suggestion
				if recreated || tryrecreated {
					newF, err := os.Open(path)
					if err != nil {
						if os.IsNotExist(err) {
						       tryrecreated = true
							continue
						}
						return fmt.Errorf("failed to open log file %q: %v", path, err)
					}
					tryrecreated = false

```

Reply:
Let me think about this.
I am thinking some other changes now.
If there are no other changes, I will update as you suggested..

Reply:
I have updated the source code as you suggested.
And also I changed value name from `tryrecreated` to `waitRecreate`.